spec/requests/tasks_spec.rb

```diff
@@ -0,0 +1,51 @@
+describe 'TasksController' do
+  let(:task) { FactoryGirl.create(:miq_task, :state => MiqTask::STATE_FINISHED) }
+  let(:task2) { FactoryGirl.create(:miq_task, :state => MiqTask::STATE_FINISHED) }
+
+  def expect_deleted(*args)
+    args.each do |arg|
+      expect(MiqTask.find_by(:id => arg.id)).to be_nil
+    end
+  end
+
+  it 'deletes on DELETE' do
+    api_basic_authorize resource_action_identifier(:tasks, :delete, :delete)
+
+    url = api_task_url(nil, task.id)
+
+    delete(url)
+
+    expect(response).to have_http_status(204)
+    expect_deleted(task)
+  end
+
+  it 'deletes on POST' do
+    api_basic_authorize resource_action_identifier(:tasks, :delete)
+
+    url = api_task_url(nil, task.id)
+    data = {:action => 'delete'}
+
+    post(url, :params => data)
+
+    expect(response).to have_http_status(200)
+    expect_deleted(task)
+  end
+
+  it 'bulk deletes' do
+    api_basic_authorize collection_action_identifier(:tasks, :delete)
+
+    url = api_tasks_url
```

Comment:
are you only using this variable in one place? May not be necessary to assign it